src/goto-symex/goto_symex_state.cpp

```diff
@@ -331,8 +336,10 @@ exprt goto_symex_statet::rename(exprt expr, const namespacet &ns)
     // we never rename function symbols
     if(as_const(expr).type().id() == ID_code)
       rename<level>(expr.type(), to_symbol_expr(expr).get_identifier(), ns);
-    else
+    else if(level == L0 || level == L1)
       expr = rename<level>(ssa_exprt(expr), ns);
+    else // No SSA version for level2
+      expr = rename<level>((exprt)ssa_exprt(expr), ns);
```

Comment:
Is this still necessary, given they ssa vs. non-ssa functions have different names? It if is, use `static_cast`, not `(exprt)`.